packages/java/scripts/download-jdt-ls.js

```diff
@@ -47,32 +51,42 @@ function downloadJavaServer() {
         const file = fs.createWriteStream(archivePath);
         const downloadWithRedirect = url => {
             /** @type { any } */
-            const h = url.toString().startsWith("https") ? https : http;
+            const h = url.toString().startsWith('https') ? https : http;
             h.get(url, response => {
                 const statusCode = response.statusCode;
                 const redirectLocation = response.headers.location;
                 if (statusCode >= 300 && statusCode < 400 && redirectLocation) {
-                    console.log("redirect location: " + redirectLocation)
                     downloadWithRedirect(redirectLocation);
                 } else if (statusCode === 200) {
-                    response.on("end", e => resolve());
-                    response.on("error", e => {
+                    response.on('end', () => resolve());
+                    response.on('error', e => {
                         file.destroy();
                         reject(e);
                     });
                     response.pipe(file);
                 } else {
                     file.destroy();
-                    reject(`failed to download with code: ${statusCode}`);
+                    reject(new Error(`Failed to download Java LS with code: ${statusCode}`));
                 }
             })
 
         };
-        downloadWithRedirect(archiveUri);
+        downloadWithRedirect(downloadUrl);
     });
 }
 
+// Just to make sure we can reverse-engineer the download URL from the hash of the file name.
```

Comment:
When do we need this info? And would a simple log file also work to just understand the download history?

Reply:
I do not have any preferences on that. Eventually, we can get rid of this download history at all.

Reply:
> I do not have any preferences on that.

Me neither. Just asking for a reason. But if that can help debugging...